tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/CiamUsernamePasswordIntegrationTests.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Linq;
+using System.Net;
+using System.Security;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client;
+using Microsoft.Identity.Client.Internal;
+using Microsoft.Identity.Test.Common;
+using Microsoft.Identity.Test.Integration.Infrastructure;
+using Microsoft.Identity.Test.Integration.net45.Infrastructure;
+using Microsoft.Identity.Test.LabInfrastructure;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Identity.Test.Integration.NetFx.HeadlessTests
+{
+    [TestClass]
+    public class CiamUsernamePasswordIntegrationTests
+    {
+        private const string _extraQParams = "dc=ESTS-PUB-EUS-AZ1-FD000-TEST1";
+        private readonly string[] _ciamScopes = new[] { "openid" };
+        private const string _ciamRedirectUri = "http://localhost";
+
+        [TestMethod]
+        [DataRow("https://{0}" + Constants.CiamAuthorityHostSuffix + "/", 0)]
+        [DataRow("https://{0}" + Constants.CiamAuthorityHostSuffix + "/{1}.onmicrosoft.com", 1)]
+        [DataRow("https://{0}" + Constants.CiamAuthorityHostSuffix + "/{1}", 2)]
```

Comment:
Side note. There's a difference in the cache item keys generated for these different authorities. For 1 and 2, cache key uses a domain tenant; while for 3, it uses a GUID tenant. This is because the tenant for this is retrieved from the authority. The partition key is home account ID for all cases, since we use client info for that. This is also the same for B2C scenario. The only scenario which this can affect that I can think of is if the cache is shared between two app instances which use different tenant formats which would result in a cache miss for the users.

cc @bgavrilMS

Reply:
Not sure we can do anything about it though, we don't have the GUID upfront in MSAL. It's the same for AAD if they choose to use friendy name instead of GUID.